models/Definition.php

```diff
@@ -204,19 +206,57 @@ protected function addItemToSet($item, $url, $mtime = null)
         return $urlSet;
     }
 
-    protected function makeUrlElement($xml, $pageUrl, $lastModified, $frequency, $priority)
+    /**
+     * Build the URL element for the provided information
+     *
+     * @param DomDocument $xml The XML object to write to
+     * @param string $pageUrl The URL to generate an item for
+     * @param string $lastModified The ISO 8601 date that the item was last modified
+     * @param DefinitionItem $item The actual definition item object
+     */
+    protected function makeUrlElement($xml, $pageUrl, $lastModified, $item)
     {
         if ($this->urlCount >= self::MAX_URLS) {
             return false;
         }
 
+        /**
+         * @event winter.sitemap.beforeMakeUrlElement
+         * Provides an opportunity to prevent an element from being produced
+         *
+         * Example usage (stops the generation process):
+         *
+         *     Event::listen('winter.sitemap.beforeMakeUrlElement', function ((Definition) $definition, (DomDocument) $xml, (string) &$pageUrl, (string) &$lastModified, (DefinitionItem) $item) {
+         *         if ($pageUrl === '/ignore-this-specific-page') {
+         *             return false;
+         *         }
+         *     });
+         *
+         */
+        if (Event::fire('winter.sitemap.beforeMakeUrlElement', [$this, $xml, &$pageUrl, &$lastModified, $item], true) === false) {
+            return false;
+        }
+
         $this->urlCount++;
 
         $url = $xml->createElement('url');
         $url->appendChild($xml->createElement('loc', $pageUrl));
         $url->appendChild($xml->createElement('lastmod', $lastModified));
-        $url->appendChild($xml->createElement('changefreq', $frequency));
-        $url->appendChild($xml->createElement('priority', $priority));
+        $url->appendChild($xml->createElement('changefreq', $item->changefreq));
+        $url->appendChild($xml->createElement('priority', $item->priority));
```

Comment:
It would make sense to rename the `$url` variable to `$urlElement` (less confusion)

Reply:
I agree. Will do so.